### ISSUE TITLE
rhash-1.3.5 won't compile due to missing -lintl on Gentoo/FreeBSD

### ISSUE THREAD

**Issue by KireinaHoro** [rhash/RHash#43]
Build log:

```
[32;01m * [39;49;00mPackage:    app-crypt/rhash-1.3.5
[32;01m * [39;49;00mRepository: gentoo
[32;01m * [39;49;00mMaintainer: sautier.louis@gmail.com proxy-maint@gentoo.org,chewi@gentoo.org
[32;01m * [39;49;00mUpstream:   https://github.com/rhash/RHash/issues
[32;01m * [39;49;00mUSE:        abi_x86_64 amd64-fbsd elibc_FreeBSD kernel_FreeBSD nls userland_BSD
[32;01m * [39;49;00mFEATURES:   preserve-libs userpriv
>>> Unpacking source...
>>> Unpacking rhash-1.3.5-src.tar.gz to /var/tmp/portage/app-crypt/rhash-1.3.5/work
>>> Source unpacked in /var/tmp/portage/app-crypt/rhash-1.3.5/work
>>> Preparing source in /var/tmp/portage/app-crypt/rhash-1.3.5/work/RHash-1.3.5 ...
 [32;01m*[0m Will copy sources from /var/tmp/portage/app-crypt/rhash-1.3.5/work/RHash-1.3.5
 [32;01m*[0m abi_x86_64.amd64_fbsd: copying to /var/tmp/portage/app-crypt/rhash-1.3.5/work/RHash-1.3.5-abi_x86_64.amd64_fbsd
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/app-crypt/rhash-1.3.5/work/RHash-1.3.5 ...
 [32;01m*[0m abi_x86_64.amd64_fbsd: running multilib-minimal_abi_src_configure
>>> Source configured.
>>> Compiling source in /var/tmp/portage/app-crypt/rhash-1.3.5/work/RHash-1.3.5 ...
 [32;01m*[0m abi_x86_64.amd64_fbsd: running multilib-minimal_abi_src_compile
gmake -j4 'CFLAGS=-O2 -pipe -mtune=generic' LDFLAGS= CC=clang 'ADDCFLAGS=-DNDEBUG -DUSE_GETTEXT' ADDLDFLAGS= PREFIX=/usr LIBDIR=$(PREFIX)/lib build-shared 
clang -c -pipe -O2 -pipe -mtune=generic -DNDEBUG -DUSE_GETTEXT -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations calc_sums.c -o calc_sums.o
clang -c -pipe -O2 -pipe -mtune=generic -DNDEBUG -DUSE_GETTEXT -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations hash_print.c -o hash_print.o
echo "#define VERSION \"1.3.5\"" > version.h
clang -c -pipe -O2 -pipe -mtune=generic -DNDEBUG -DUSE_GETTEXT -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations hash_update.c -o hash_update.o
clang -c -pipe -O2 -pipe -mtune=generic -DNDEBUG -DUSE_GETTEXT -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations file_mask.c -o file_mask.o
clang -c -pipe -O2 -pipe -mtune=generic -DNDEBUG -DUSE_GETTEXT -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations file_set.c -o file_set.o
clang -c -pipe -O2 -pipe -mtune=generic -DNDEBUG -DUSE_GETTEXT -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations find_file.c -o find_file.o
clang -c -pipe -O2 -pipe -mtune=generic -DNDEBUG -DUSE_GETTEXT -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations hash_check.c -o hash_check.o
clang -c -pipe -O2 -pipe -mtune=generic -DNDEBUG -DUSE_GETTEXT -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations output.c -o output.o
clang -c -pipe -O2 -pipe -mtune=generic -DNDEBUG -DUSE_GETTEXT -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations parse_cmdline.c -o parse_cmdline.o
clang -c -pipe -O2 -pipe -mtune=generic -DNDEBUG -DUSE_GETTEXT -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations rhash_main.c -o rhash_main.o
clang -c -pipe -O2 -pipe -mtune=generic -DNDEBUG -DUSE_GETTEXT -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations win_utils.c -o win_utils.o
gmake -C librhash lib-shared
gmake[1]: Entering directory `/var/tmp/portage/app-crypt/rhash-1.3.5/work/RHash-1.3.5-abi_x86_64.amd64_fbsd/librhash'
sed -n '1s/.*/{ global:/p; s/^RHASH_API.* \([a-z0-9_]\+\)(.*/  \1;/p; $s/.*/local: *; };/p' rhash.h rhash_torrent.h rhash_timing.h > exports.sym
clang -c -pipe -O2 -pipe -mtune=generic -DNDEBUG -DUSE_GETTEXT -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations common_func.c -o common_func.o
clang -fpic  -pipe -DIN_RHASH -DNDEBUG -DUSE_GETTEXT -O2 -pipe -mtune=generic -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations -shared algorithms.c byte_order.c plug_openssl.c rhash.c rhash_timing.c rhash_torrent.c aich.c crc32.c ed2k.c edonr.c hex.c md4.c md5.c sha1.c sha256.c sha512.c sha3.c ripemd-160.c gost.c has160.c snefru.c tiger.c tiger_sbox.c tth.c torrent.c whirlpool.c whirlpool_sbox.c -Wl,--version-script,exports.sym,-soname,librhash.so.0   -o librhash.so.0
rm -f librhash.so
ln -s librhash.so.0 librhash.so
gmake[1]: Leaving directory `/var/tmp/portage/app-crypt/rhash-1.3.5/work/RHash-1.3.5-abi_x86_64.amd64_fbsd/librhash'
clang calc_sums.o hash_print.o common_func.o hash_update.o file_mask.o file_set.o find_file.o hash_check.o output.o parse_cmdline.o rhash_main.o win_utils.o   -o rhash_shared librhash/librhash.so.0
calc_sums.o: In function `rename_file_by_embeding_crc32':
calc_sums.c:(.text+0xbf): undefined reference to `libintl_gettext'
calc_sums.c:(.text+0x2d1): undefined reference to `libintl_gettext'
calc_sums.o: In function `save_torrent_to':
calc_sums.c:(.text+0x516): undefined reference to `libintl_gettext'
calc_sums.o: In function `check_hash_file':
calc_sums.c:(.text+0xc8b): undefined reference to `libintl_gettext'
calc_sums.c:(.text+0xd01): undefined reference to `libintl_gettext'
calc_sums.o:calc_sums.c:(.text+0x1124): more undefined references to `libintl_gettext' follow
rhash_main.o: In function `main':
rhash_main.c:(.text+0x113): undefined reference to `libintl_bindtextdomain'
rhash_main.c:(.text+0x11d): undefined reference to `libintl_textdomain'
rhash_main.c:(.text+0x180): undefined reference to `libintl_gettext'
rhash_main.c:(.text+0x1d9): undefined reference to `libintl_gettext'
rhash_main.c:(.text+0x1ed): undefined reference to `libintl_gettext'
rhash_main.c:(.text+0x2b8): undefined reference to `libintl_gettext'
rhash_main.c:(.text+0x2f8): undefined reference to `libintl_gettext'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
gmake: *** [rhash_shared] Error 1
 [31;01m*[0m ERROR: app-crypt/rhash-1.3.5::gentoo failed (compile phase):
 [31;01m*[0m   emake failed
 [31;01m*[0m 
 [31;01m*[0m If you need support, post the output of `emerge --info '=app-crypt/rhash-1.3.5::gentoo'`,
 [31;01m*[0m the complete build log and the output of `emerge -pqv '=app-crypt/rhash-1.3.5::gentoo'`.
 [31;01m*[0m The complete build log is located at '/var/tmp/portage/app-crypt/rhash-1.3.5/temp/build.log'.
 [31;01m*[0m The ebuild environment file is located at '/var/tmp/portage/app-crypt/rhash-1.3.5/temp/environment'.
 [31;01m*[0m Working directory: '/var/tmp/portage/app-crypt/rhash-1.3.5/work/RHash-1.3.5-abi_x86_64.amd64_fbsd'
 [31;01m*[0m S: '/var/tmp/portage/app-crypt/rhash-1.3.5/work/RHash-1.3.5'
```

Building with `make ADDLDFLAGS="-lintl"` works.

**Comment by rhash** [comment]
The linker options depend on linux distribution and on the standart library (libc).

For example, Debian/Ubuntu Linux doesn't contain separate libintl library. Libintl functions (and headers) are provided by the libc6 package (glibc) and are linked automatically.

So, maintainers of rhash package must pick linker options specifically for each Distribution.

**Comment by Skif-off** [comment]
@KireinaHoro you can compare ```Makefile```(s) with [1.3.4](https://svnweb.freebsd.org/ports/head/security/rhash/) and try patches from "files".

**Comment by rhash** [comment]
I think correct liker options should be detected by a configure script, and I'm working on it now ;)

**Comment by rhash** [comment]
@KireinaHoro compilation issue fixed by commit cdaf43862229fb0f10d2b6b7448bcded2ef8edde

**Comment by KireinaHoro** [comment]
Seems great. It works fine now.

2018年1月29日(月) 午前6:11 Aleksey Kravchenko <notifications@github.com>:
